### PR TITLE
PE-2 Add initial workflows

### PR DESCRIPTION
**Pull request by ryan-breidenbach**
This is the first pass at adding shared workflow for building and deploying TypeScript Lambda applications. We have quite a few of these applications now, so we should consolidate and standardize our workflows. This PR adds:

1. Workflow for building a release candidate
2. Workflow for deploying a release candidate to AWS
3. [README](https://github.com/invitation-homes/typescript-lambda-workflows/tree/PE-2-add-initial-workflows) for this repository

### CODE REVIEW

File: .github/workflows/deploy-application.yml [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    needs: upload-lambda-function
+    steps:
+
+    - name: Checkout code
+      uses: actions/checkout@v3
+      with:
+        ref: ${{ inputs.tag }}
+
+    - name: Setup Terraform
+      uses: hashicorp/setup-terraform@v1
```
Comment (FooBartn) [inline .github/workflows/deploy-application.yml:133]:
```suggestion
      uses: hashicorp/setup-terraform@v2
```

File: .github/workflows/build-release-candidate.yml [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+      version: ${{ steps.generate-version.outputs.version }}
+      upload_url: ${{ steps.create-release.outputs.upload_url }}
+
+  generate-function-list:
+    name: Generate list of Lambda functions
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Checkout code
+      uses: actions/checkout@v2
```
Comment (FooBartn) [inline .github/workflows/build-release-candidate.yml:51]:
```suggestion
      uses: actions/checkout@v3
```

File: .github/workflows/build-release-candidate.yml [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  workflow_call
+
+jobs:
+  create-github-release:
+    name: Create GitHub Release
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Checkout code
+      uses: actions/checkout@v2
```
Comment (FooBartn) [inline .github/workflows/build-release-candidate.yml:13]:
```suggestion
      uses: actions/checkout@v3
```

File: .github/workflows/build-release-candidate.yml [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    needs: [create-github-release, generate-function-list]
+    strategy:
+      matrix:
+        function: ${{ fromJson(needs.generate-function-list.outputs.functions) }}
+    env:
+      ZIP_FILE_NAME: ${{matrix.function}}-${{needs.create-github-release.outputs.version}}.zip
+    steps:
+
+    - name: Check out code
+      uses: actions/checkout@v2
```
Comment (FooBartn) [inline .github/workflows/build-release-candidate.yml:77]:
```suggestion
      uses: actions/checkout@v3
```

File: .github/workflows/build-release-candidate.yml [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+    - name: Install Node.js v16
+      uses: actions/setup-node@v2
```
Comment (FooBartn) [inline .github/workflows/build-release-candidate.yml:80]:
```suggestion
      uses: actions/setup-node@v3
```

File: .github/workflows/deploy-application.yml [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+    - name: Get release
+      run: >
+        curl --fail -s
+        -H 'authorization: Bearer ${{ secrets.GITHUB_TOKEN }}'
+        -H 'content-type: application/json'
+        ${{ env.BASE_API_URL }}/tags/${{ inputs.tag }}
+        > release.json
+
+    - name: Upload release details as artifact
+      uses: actions/upload-artifact@v2
```
Comment (FooBartn) [inline .github/workflows/deploy-application.yml:56]:
```suggestion
      uses: actions/upload-artifact@v3
```

File: .github/workflows/deploy-application.yml [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+      with:
+        aws-access-key-id: ${{ secrets.DEPLOYER_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.DEPLOYER_SECRET_ACCESS_KEY }}
+        aws-region: us-east-1
+        role-to-assume: arn:aws:iam::${{ steps.look-up-aws-account-id.outputs.account-id }}:role/github-actions-deployer
+        role-skip-session-tagging: true
+        role-duration-seconds: 1200
+
+    - name: Download release details artifact
+      uses: actions/download-artifact@v2
```
Comment (FooBartn) [inline .github/workflows/deploy-application.yml:99]:
```suggestion
      uses: actions/download-artifact@v3
```

**Review by FooBartn — DISMISSED** [review]
Non-blocking changes suggested for latest action versions

**Review by edwagner — APPROVED** [review]
I didn't exhaustively vet the details, but this looks great.

**Pull request merged**